### ISSUE TITLE
Underfitting multi-task model

### ISSUE THREAD

**Issue by prajwalkr** [facebookresearch/ELI5#22]
**Bug description**
I am trying to train the multi-task model on the given dataset.  I am facing the issue of under-fitting (see logs below) and am able to achieve a valid ppl of just 59, and train ppl of 70, both of which have plateaued for quite some time now. 

**Reproduction steps**
The dataset is also prepared from the `process_data_source_target.py` file. I have skipped the masking task for now. LM + Seq2Seq is what I am trying to train. All other hyper-parameters are same except: `max-source-length` `max-target-length` is `1024`. To fit into my available GPU memory, the maximum effective batch-size I could achieve was 24 (this too with a smaller Transformer model `transformer_wmt_en_de`).

**Expected behavior**
The paper reports ppl of around 33. I hope to achieve this, was able to achieve the paper's PPL with the Q+A model. 

**Logs**
Please paste the command line output:
2020-04-19 10:29:42 | INFO | valid | epoch 009 | valid on 'valid' subset | loss 6.604 | nll_loss 5.901 | ppl 59.76 | wps 45030.8 | wpb 1528.7 | bsz 12 | num_updates 406000 | best_loss 6.604
2020-04-19 10:29:56 | INFO | fairseq.checkpoint_utils | saved checkpoint checkpoint_9_406000.pt (epoch 9 @ 406000 updates, score 6.604) (writing took 13.
965450627729297 seconds)
2020-04-19 10:30:21 | INFO | train_inner | epoch 009:  10798 / 52601 loss=6.808, nll_loss=6.135, ppl=70.27, wps=17040.2, ups=2, wpb=8512.8, bsz=24, num_updates=406100, lr=9.92461e-06, gnorm=1.384, loss_scale=64, train_wall=24, wall=0

**Additional context**
Could you suggest ANY "hacks" (for lack of a better word) given my computational limitations as to how I can get close to the paper's scores on multi-task model?

**Comment by huihuifan** [comment]
hello, are you training a model of the same size? It looks like you are using a smaller model?

If you need to decrease the batch size, you can "increase" it back to the usual size by manipulating the --update-freq parameter in fairseq. --update-freq accumulates gradients to simulate training on a larger batch. I wonder if you have optimization challenges - as the learning rate and batch size affect each other a lot, it is important to modify those together. 

For the paper, we did train longer than 9 epochs, but please check the optimization parameters if you report performance plateau-ing.

**Comment by prajwalkr** [comment]
> hello, are you training a model of the same size? It looks like you are using a smaller model?

Yes, as I have a limited memory. If I use the bigger model, then even after using update-freq the effective batch-size is 24 or so and it is extremely slow as well.  

> If you need to decrease the batch size, you can "increase" it back to the usual size by manipulating the --update-freq parameter in fairseq. --update-freq accumulates gradients to simulate training on a larger batch. I wonder if you have optimization challenges - as the learning rate and batch size affect each other a lot, it is important to modify those together.

Can you specify the effective batch-size used for the paper? I will try to get close to that. And what is the learning rate schedule corresponding to this?
